src/utils/validation.ts

```diff
@@ -10,8 +10,10 @@ export function checkArray (array: Tensor | RecursiveArray<number>, dtype: keyof
     if (ensureDimension != -1 && dim != ensureDimension){
       throw new TypeError(`Dimension of input require to be ${ensureDimension} but receive ${dim}`);
     }
-    if (dtype != arr_dtype){
-      array = cast(array, dtype);
+    if (dtype == 'string' || dtype == 'float32' || dtype == 'bool' || dtype == 'int32' || dtype == 'complex64'){
```

Comment:
```suggestion
    if (dtype === 'string' || dtype === 'float32' || dtype === 'bool' || dtype === 'int32' || dtype === 'complex64') {
```
Could we define an enum for the data type?